include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -405,6 +406,23 @@ InternalRouteResult extractRoute(const DataFacade<AlgorithmT> &facade,
     return raw_route_data;
 }
 
+template <typename FacadeT> EdgeDistance computeEdgeDistance(const FacadeT &facade, NodeID node_id)
+{
+    const auto geometry_index = facade.GetGeometryIndex(node_id);
+
+    EdgeDistance total_distance = 0.0;
+
+    auto geometry_range = facade.GetUncompressedForwardGeometry(geometry_index.id);
+    for (auto current = std::next(geometry_range.begin()); current != geometry_range.end();
+         ++current)
+    {
+        total_distance += util::coordinate_calculation::haversineDistance(
+            facade.GetCoordinateOfNode(*std::prev(current)), facade.GetCoordinateOfNode(*current));
+    }
```

Comment:
we can also change the for loop just like the others such that it will be 

`total_distance += util::coordinate_calculation::haversineDistance(facade.GetCoordinateOfNode(*current), facade.GetCoordinateOfNode(*std::next(current)));`